### PR TITLE
Generic text field

### PR DESCRIPTION
**Pull request by nicopaul**
Added all functionalities of the PageObject GenericTextField:
-PageObject GenericTextField
-GenericTextFieldTest
-IntegrationTest
-Assert
-AssertTest
-Matcher
-MatcherTest

---
This change is [Reviewable](https://reviewable.io/reviews/testit-webtester/webtester-core/61)

### CODE REVIEW

**Comment by bitterblue** [comment]
Looks good to me :)
Can you please update the documentation (`support-hamcrest.md` and `support-assertj.md`)?

File: webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java [hunk 1]
```diff
@@ -54,6 +55,10 @@ protected String getHTMLFilePath() {
 
     // UTILITY
 
+    protected void assertPageObjectCanBeInitialized(PageObject object) {
```
Comment (bitterblue) [inline webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java]: I cannot find any usage of this assertion within the PR. Why did you add it?

Comment (nicopaul) [inline webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java]: I need this assertion for other PageObjects which are based on GenericTextField. 
That is why I added it on this branch, so I dont have to add it on all the other Branches.

Comment (bitterblue) [inline webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java]: I looked at some of the branches in your repo (EmailField, UrlField, SearchField) and didn't find any references to the method. Can you point me to a branch, where you are actually using it? I don't want to integrate dead code.

Comment (bitterblue) [inline webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java]: Thanks!
I'll keep this commit out, but accept the rest of your pull request. You can add the commit to a future PR where the test method is used.

File: webtester-support-hamcrest/src/main/java/info/novatec/testit/webtester/support/hamcrest/GenericTextFieldMatcher.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+
+/**
+ * Contains {@link Matcher matcher} for {@link GenericTextField generic text fields}.
+ *
+ * @since 1.2
+ */
+public class GenericTextFieldMatcher extends PageObjectMatcher {
```
Comment (bitterblue) [inline webtester-support-hamcrest/src/main/java/info/novatec/testit/webtester/support/hamcrest/GenericTextFieldMatcher.java]: And one more thing: please turn the constructor private (instead of protected), so that codacy is happy?

File: webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java (hunk 1 above)
Comment (nicopaul) [inline webtester-core/src/test/java/integration/AbstractWebTesterIntegrationTest.java]: Okay

**Comment by bitterblue** [comment]
Nico, I merged your backport of `GenericTextField`. Thanks, and don't forget to pull the current master branch!
